run.py

```diff
@@ -418,6 +478,80 @@ def _synthesis_morphing(
 
         return FileResponse(f.name, media_type="audio/wav")
 
+    @app.post(
+        "/guided_synthesis",
+        responses={
+            200: {
+                "content": {
+                    "audio/wav": {"schema": {"type": "string", "format": "binary"}}
+                },
+            }
+        },
+        tags=["音声合成"],
+        summary="Audio synthesis guided by external audio and phonemes",
+    )
+    def guided_synthesis(
```

Comment:
`guided synthesis`機能はまだ実験的な段階で、APIが変わる可能性があると思います。
例えば`extract_guided_feature`は２回実行しなくても良いはず。

実験的な段階であることがわかるように、エンジン起動時に実験フラグを用意するのはどうでしょう？
`enable_cancellable_synthesis`のような引数を考えています。

---

The `guided synthesis` feature is still experimental and the API may change.
For example, `extract_guided_feature` should not have to be run twice.

How about providing an experimental flag at engine startup to indicate that it is in an experimental phase?
I am thinking of an argument like `enable_cancellable_synthesis`.
https://github.com/VOICEVOX/voicevox_engine/blob/3fdaf2bd6376cfc980759b59599d225ee6a9d5a1/run.py#L695

Reply:
> 例えばextract_guided_featureは２回実行しなくても良いはず。

How? The two APIs each calls it only once.

> 実験的な段階であることがわかるように、エンジン起動時に実験フラグを用意するのはどうでしょう？

I don't know... I've waited 2 months and I'm not sure if doing this would make the waiting time longer. I tend to put it into use first, then debug or improve the details. After all I'm half done with the GUI part.

Reply:
> How? The two APIs each calls it only once.

それぞれ目的が異なることを理解しました。実行はそれぞれで必要だと思います。
I understood that each of them has different purposes. I know that implementation is necessary in each.

> I've waited 2 months and I'm not sure if doing this would make the waiting time longer.

お待たせしてしまったのは申し訳ないです･･･。

今は実験的機能にしておけば、大雑把でバグがあってもユーザーは許してくれるはずです。
結果的にそちらのほうが手っ取り早く実装できると思います。

`enable_cancellable_synthesis`を真似すれば実装できると思います。

---

Sorry to keep you waiting....

If we keep it as an experimental feature for now, users will allow it even if it is sketchy and buggy.
As a result, I think it would be quicker to implement that way.

I think you could implement it by copying `enable_cancellable_synthesis`.